File: grade_system.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

string grade_category(int g) {
    if (g < 50) return "Fail";
    else if (g < 70) return "Pass";
    else if (g < 80) return "Credit";
    else if (g < 90) return "Distinction";
    else return "High Distinction";
}

int main() {
    vector<int> grades = {45, 67, 72, 85, 93, 58, 79, 88, 100};
    int fail=0, pass=0, credit=0, dist=0, high=0;
    for (int g : grades) {
        string cat = grade_category(g);
        cout << g << " -> " << cat << endl;
        if (cat == "Fail") fail++;
        else if (cat == "Pass") pass++;
        else if (cat == "Credit") credit++;
        else if (cat == "Distinction") dist++;
        else high++;
    }
    cout << "Summary: F=" << fail << " P=" << pass << " C=" << credit
         << " D=" << dist << " HD=" << high << endl;
    return 0;
}

File: grade_system.py
def grade_category(g):
    if g < 50:
        return "Fail"
    elif g < 70:
        return "Pass"
    elif g < 80:
        return "Credit"
    elif g < 90:
        return "Distinction"
    else:
        return "High Distinction"

grades = [45, 67, 72, 85, 93, 58, 79, 88, 100]
fail = pass_ = credit = dist = high = 0

for g in grades:
    cat = grade_category(g)
    print(g, "->", cat)
    if cat == "Fail": fail += 1
    elif cat == "Pass": pass_ += 1
    elif cat == "Credit": credit += 1
    elif cat == "Distinction": dist += 1
    else: high += 1

print(f"Summary: F={fail} P={pass_} C={credit} D={dist} HD={high}")
